### PR TITLE
Support hook dependencies

### LINKED ISSUE #296: Please add the ability to define hook dependencies
### Description

Auth0 hooks allow the addition of dependencies which are NPM modules that can be used in the javascript.  It would be great if we could set the dependencies via the `auth0_hook` resource.

### New or Affected Resource(s)

* auth0_hook

### Potential Terraform Configuration

```hcl
resource "auth0_hook" "send_password_reset_email" {
  enabled    = true
  name       = "send-password-reset-email"
  trigger_id = "post-user-registration"

   dependencies = {
       auth0 = "2.30.0"
   }

  script = ....
}
```

### References

The management API supports passing dependencies: https://auth0.com/docs/api/management/v2/#!/Hooks/post_hooks

The json snippit for dependencies should look like: 
```
"dependencies": {
      "auth0": "2.30.0"
    },
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by woz5999**
### Proposed Changes

* Add support for hook "dependencies"

Fixes #296

#### Acceptance Test Output

```
$make testacc TESTS=TestAccHook
==> Checking that code complies with gofmt requirements...
?       github.com/alexkappa/terraform-provider-auth0   [no test files]
=== RUN   TestAccHook
--- PASS: TestAccHook (5.46s)
PASS
coverage: 7.9% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0     5.793s  coverage: 7.9% of statements
?       github.com/alexkappa/terraform-provider-auth0/auth0/internal/debug      [no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/random     0.277s  coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok      github.com/alexkappa/terraform-provider-auth0/auth0/internal/validation 0.266s  coverage: 0.0% of statements [no tests to run]
?       github.com/alexkappa/terraform-provider-auth0/version   [no test files]
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by woz5999** [comment]
@alexkappa are you able to review/merge this?

**Comment by woz5999** [comment]
@alexkappa could you review?

**Comment by woz5999** [comment]
@alexkappa would love to get this reviewed and merged

File: auth0/resource_auth0_hook_test.go [hunk 1]
```diff
@@ -53,6 +54,9 @@ resource "auth0_hook" "my_hook" {
   trigger_id = "pre-user-registration"
   script = "function (user, context, callback) { console.log(user); callback(null, { user }); }"
   enabled = false
+	dependencies = {
```
Comment (yvovandoorn) [inline auth0/resource_auth0_hook_test.go:57]: It looks like the formatting is off? (tabs v space?)

**Comment by yvovandoorn** [comment]
Hi @woz5999! Could you resolve the merge conflict that was introduced when #262  was merged in? 

Both are making updates to the hook test and #262 added quite a few new lines of test code. 

I can merge following the updates.

**Comment by woz5999** [comment]
@yvovandoorn this should be good now.

**Comment by yvovandoorn** [comment]
Hey @woz5999. This isn't passing my local testing setup:

```
make testacc TESTS=TestAccHookSecrets                          
==> Checking that code complies with gofmt requirements...
?   	github.com/alexkappa/terraform-provider-auth0	[no test files]
=== RUN   TestAccHookSecrets
    testing.go:705: Step 0 error: Check failed: Check 2/7 error: auth0_hook.my_hook: Attribute 'dependencies.auth0' not found
--- FAIL: TestAccHookSecrets (1.64s)
FAIL
coverage: 8.1% of statements
FAIL	github.com/alexkappa/terraform-provider-auth0/auth0	1.667s
?   	github.com/alexkappa/terraform-provider-auth0/auth0/internal/debug	[no test files]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok  	github.com/alexkappa/terraform-provider-auth0/auth0/internal/random	0.023s	coverage: 0.0% of statements [no tests to run]
testing: warning: no tests to run
PASS
coverage: 0.0% of statements
ok  	github.com/alexkappa/terraform-provider-auth0/auth0/internal/validation	0.018s	coverage: 0.0% of statements [no tests to run]
?   	github.com/alexkappa/terraform-provider-auth0/version	[no test files]
FAIL
make: *** [testacc] Error 1
```

It looks like 
```
resource.TestCheckResourceAttr("auth0_hook.my_hook", "dependencies.auth0", "2.30.0"),
```
should be added to L64 in `auth0/resource_auth0_hook_test.go`

and 
```
  dependencies = {
    auth0 = "2.30.0"
  }
```
to L93-95 in `auth0/resource_auth0_hook_test.go`

This way it passes the initial and diff check.

**Comment by woz5999** [comment]
@yvovandoorn sorry about that. got those added.

**Comment by yvovandoorn** [comment]
👍 LGTM

**Comment by galkin** [comment]
@yvovandoorn, do you have new release estimate?

**Comment by yvovandoorn** [comment]
@galkin probably in the coming days. I hope to incorporate #321 but its dependent on https://github.com/go-auth0/auth0/pull/187. That said, if its taking too long, I'll talk with @alexkappa to push out a release earlier. 

Thanks for your patience on this!!

**Comment by yvovandoorn** [comment]
@galkin released! https://github.com/alexkappa/terraform-provider-auth0/releases/tag/v0.20.0

**Pull request merged**